### PR TITLE
✨ Added mypy type checks + support any iterable

### PR DESCRIPTION
**Pull request by JoaquimEsteves**
Type checking is neat.

Added all the necessary config/setup files and added type hinting to `mail.py`.

send_mail now also supports any iterable instead of just dictionaries.

### CODE REVIEW

File: justsendmail/mail.py [hunk 1]
```diff
@@ -60,15 +62,29 @@ def send_mail(
...
+                encoders.encode_base64(part)
+                part.add_header(
+                    "Content-Disposition",
+                    'attachment; filename="{}"'.format(os.path.basename(k)),
+                )
+                msg.attach(part)
+        else:
+            # Assume it's a general iterable
+            for attachment in attachments:
+                part = MIMEBase("application", "octet-stream")
```
Comment (pschmitt) [inline justsendmail/mail.py:78]: DRY

File: justsendmail/mail.py [hunk 2]
```diff
@@ -10,17 +10,19 @@
 import argparse
 import logging
 import os
-from typing import Optional, Dict
+from typing import Optional, Iterable, Dict, Union, List
```
Comment (pschmitt) [inline justsendmail/mail.py:13]: I'd prefer to see `import typing` here.

File: justsendmail/mail.py (hunk 1 above)
Comment (JoaquimEsteves) [reply justsendmail/mail.py]: Done in 704e60a087ee476d3c04288bfc0b800766c85a6e

Will test it tomorrow

File: justsendmail/mail.py (hunk 2 above)
Comment (JoaquimEsteves) [reply justsendmail/mail.py]: `import typing as t`

704e60a087ee476d3c04288bfc0b800766c85a6e

**Comment by pschmitt** [comment]
Lgtm at first glance. Please remind me tomorrow to review this properly.

**Pull request merged**